hierarchy.el

```diff
@@ -535,10 +570,22 @@ node label."
   (require 'wid-edit)
   (require 'tree-widget)
   (hierarchy-map-tree (lambda (item indent children)
-                        (widget-convert
-                         'tree-widget
-                         :tag (hierarchy-labelfn-to-string labelfn item indent)
-                         :args children))
+                        (let ((childrenfn (map-elt
+                                           (hierarchy--delaying-parents hierarchy)
+                                           item)))
+                          (if childrenfn
```

Comment:
would it be possible to have it the other way around? Something like:

```emacs
(widget-convert 'tree-widget ... (if childrenfn ... ...))
```

This might require using `apply`, e.g.,

```emacs
(apply #'widget-convert ... (if childrenfn ... ...))
```

Maybe always using `hierarchy--create-delayed-tree-widget` even if the hierarchy is not delayed would help?